### PR TITLE
(PA-6267) Update build_defaults.yaml to enable Amazon Linux2 7.x

### PR DESCRIPTION
**Pull request by imaqsood**

### CODE REVIEW

**Comment by joshcooper** [comment]
Please wait to merge this until we've validated amazon linux 2 in adhoc https://github.com/puppetlabs/ci-job-configs/pull/9540#issuecomment-2136639505

**Comment by joshcooper** [comment]
puppet-agent#7.x passed adhoc. This PR is blocked on adding this platform as a build and test target to the 7.x pipeline in CJC. Once that change is merged, then we can merge this without breaking the build.

**Comment by imaqsood** [comment]
> puppet-agent#7.x passed adhoc. This PR is blocked on adding this platform as a build and test target to the 7.x pipeline in CJC. Once that change is merged, then we can merge this without breaking the build.

test and build target for 7.x merged in [9543](https://github.com/puppetlabs/ci-job-configs/pull/9543)

**Review by joshcooper — APPROVED** [review]

**Pull request merged**
